toml.md

```diff
@@ -104,27 +104,44 @@ Keys
 
 A key may be either bare, quoted, or dotted.
 
-**Bare keys** may only contain ASCII letters, ASCII digits, underscores, and
-dashes (`A-Za-z0-9_-`). Note that bare keys are allowed to be composed of only
-ASCII digits, e.g. `1234`, but are always interpreted as strings.
+**Bare keys** may contain any letter-like or number-like Unicode character from
+any Unicode script, as well as ASCII digits, dashes and underscores. 
+Punctuation, spaces, arrows, box drawing and private use characters are not
+allowed.  Note that bare keys are allowed to be composed of only ASCII digits,
+e.g. `1234`, but are always interpreted as strings.
+
+* From the ASCII characters, only `A-Z`, `a-z`, `0-9`, `_` and `-` are allowed.
+* From the rest of the ISO-8859-1 characters, allowed are U+B1 to U+B3, U+B9, 
+  U+BC to U+BE, U+C0 to U+D6, U+D8 to U+F6 and U+F8 to U+FF
+* All of U+0100-1FFF are allowed, except `;` (U+037E).
+* Characters U+200C, U+200D, U+203F, U+2040, U+2070 to U+218F, U+2460 to 
+  U+24FF, and U+2C00 to U+2FEF are allowed.
+* All characters from U+3001 and higher, except surrogates (U+D800 to U+D999),
+  Private Use (U+E000 to U+F8FF, U+F0000 to U+100000) and process-internal use
+  (U+FDD0 to U+FDEF).
```

Comment:
What's with U+FFFE (65534) and U+FFFF (65535)? They are excluded by the ABNF but, as far as I can see, allowed by the text.

Also, the upper limit of the private use area are different in the text (U+100000) vs. the ABNF comment (x10FFFF). I suppose it doesn't matter because Unicode effectively ends at x10FFFF, but it's still a bit confusing. (One could just say "and above" instead.)

Reply:
@ChristianSi, the part about U+FFFE/FFFF is done. Let me know if you want to change the sentences to be inclusive all the way, except for exclusive in some cases as it is now.

Reply:
Ah, I see now what you mean. Yes, I seem to have been a bit hand-wavy to make the comments not to cumbersome and the ranges easier understood. Apparently that leads to confusion. Private use are, officially, U+F0000..U+FFFFD and U+100000..U+10FFFD (codes ending on FFFE/FFFF are forbidden and considered non-characters here as well).

I think these are little known facts and we can leave it as is, right?

The range in the text is clearly a mistake, 100000 should be 10FFFF, thanks. Sorry for misreading that in your first comment.

Reply:
Fixed. Please reopen if you think I should be more precise in the PUP ranges.